.github/workflows/docs-ci.yml

```diff
@@ -6,7 +6,40 @@ on:
   push:
     branches: [ 'main', 'release/v*' ]
 
+env:
+  version-path: ${{ startsWith(github.ref, 'refs/heads/release/v') && github.ref_name || 'next' }}
+
 jobs:
+  symlink:
+    runs-on: ubuntu-22.04
+    concurrency:
+      group: symlink-${{ github.workflow }}-${{ github.ref }}
+      # We don't want to cancel in-progress jobs against main because that might leave the upload in a bad state.
+      cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Make Version Symlinks
+        if: ${{ startsWith(env.version-path, 'release/v') && github.event_name == 'push' }}
+        run: |
+          mkdir tmp-deephaven-core; cd tmp-deephaven-core
+          ln -s ${{ env.version-path }} latest
+          # Make symlinks to preserve old bookmarks
+          ln -s next/javadoc/ javadoc; ln -s next/pydoc/ pydoc
+          ln -s next/client-api/ client-api; ln -s next/r/ r
+          
+      - name: Deploy Symlinks
+        if: ${{ startsWith(env.version-path, 'release/v') && github.event_name == 'push' }}
+        uses: burnett01/rsync-deployments@5.2
+        with:
+          switches: -rlptDvz --force --mkpath
+          path: tmp-deephaven-core/
+          remote_path: deephaven-core/
```

Comment:
I'm a bit scared about rsyncing w/ symlinks into a root directory. Not that I this is necessarily know if there is anything wrong w/ how it is expressed now, but I imagine things could go wrong w/ a mistype. Maybe we want an additional subdirectory? `deephaven-core/symlinks/`?

Reply:
Rsyncing links to the _deephaven-core_ directory seems no different than rsyncing _javadoc_ or _pydoc_ dirs into the same directory.  How are links more dangerous than those overwriting directories?  (Also, there's no _--delete_ on this deploy.)

The other issue is that for backward compatibility, at least until the deephaven.io hrefs change, the_deephaven-core/javadoc_ et al dirs are replaced with symlinks in the _next_ directory.

By the way, you probably already know this, but the _deephaven-core/_ directory is a subdirectory under a user that contains only Docs CI deployed api docs.  We aren't affecting anything outside it.

Reply:
We don't _need_ to have this change be seamless / atomic for deephaven.io. For example, we _could_ consider leaving `docs-ci.yml` unchanged, and create `docs-ci-v2.yml`, and placing everything under the remote_path `deephaven-core-docs-ci-v2` or something like that. It's hard for me to universally reason about what rsync may or may not do in certain circumstances. In fact, we may want to do real tests with a _different_ remote_path just to make sure it all seems to work.